### PR TITLE
CHORE: add NuGet release

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add NuGet release

Relates to #63

### CODE REVIEW

File: build/nuget-release.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - task: DotNetCoreInstaller@0
+            displayName: 'Import .NET Core SDK ($(DotNet.Sdk.Version))'
+            inputs:
+              version: '$(DotNet.Sdk.Version)'
+          - task: qetza.replacetokens.replacetokens-task.replacetokens@3
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Please use our YAML template for this

File: build/nuget-release.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        displayName: 'Run unit tests'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - task: DotNetCoreInstaller@0
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Please use our YAML template for this

File: build/nuget-release.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Security.All/$(Build.BuildNumber))
+                ```shell
+                PM > Install-Package Arcus.Security.All --Version $(Build.BuildNumber)
+                ```
+              isPreRelease: true
+              compareWith: 'lastRelease'
+            condition: contains(variables['Build.BuildNumber'], '-')
+          - task: NuGetCommand@2
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Let's review, merge and use https://github.com/arcus-azure/azure-devops-templates/pull/8

File: build/nuget-release.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+              tag: 'v$(Build.BuildNumber)'
+              title: 'v$(Build.BuildNumber)'
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Security.All/$(Build.BuildNumber))
+                ```shell
+                PM > Install-Package Arcus.Security.All --Version $(Build.BuildNumber)
+                ```
+            condition: not(contains(variables['Build.BuildNumber'], '-'))
+          - task: GitHubRelease@0
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Let's use our new template

File: build/nuget-release.yml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        displayName: 'Push to NuGet.org'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - task: GitHubRelease@0
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Let's use our new template

File: build/nuget-release.yml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - template: github/create-release.yml@templates
+            parameters:
+              repositoryName: 'arcus-azure/arcus.security'
+              tag: 'v$(Build.BuildNumber)'
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Adding note similar to other PR as we might want to make this the default

**Comment by tomkerkhove** [comment]
LGTM - Can you create the build afterwards please?

**Comment by stijnmoreels** [comment]
> LGTM - Can you create the build afterwards please?

Yep, will do.

File: build/nuget-release.yml [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+          - ${{ if not(contains(variables['Build.BuildNumber'], '-')) }}:
+              - template: github/create-release.yml@templates
+                parameters:
+                  repositoryName: 'arcus-azure/arcus.security'
+                  releaseNotes: |
+                    Install new version via [NuGet](https://www.nuget.org/packages/Arcus.Security.All/$(Build.BuildNumber))
+                    ```shell
+                    PM > Install-Package Arcus.Security.All --Version $(Build.BuildNumber)
+                    ```
+          - ${{ if contains(variables['Build.BuildNumber'], '-') }}:
```
Comment (tomkerkhove) [inline build/nuget-release.yml]: Hm, why did we get rid of the condition approach? :thinking:

Comment (tomkerkhove) [reply build/nuget-release.yml]: Apparently it doesn't work when calling a template

**Pull request merged**